### PR TITLE
Never propagate a WebApplicationException from the client

### LINKED ISSUE #19488: Either RESTEasy Reactive/REST Client Reactive swallow errors in some scenarios, very difficult to debug
### Describe the bug

The actual error being throw under the hood was:

```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.HashSet<org.acme.CurrentWeather>` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: (ByteArrayInputStream); line: 1, column: 1]
```

But between the fact that Quarkus generates a lot of code, and the reactive/multithreaded bit, debugging this and trying to navigate the source + callstack was somewhat of a nightmare because:
- For a lot of the source, there's no file it would show in the IDE for the relevant code bits 
- The chain of calls/indirection can be particularly deep

quarkus-reactive-not-surfacing-errors

But with this code/endpoint:
```java
@JsonIgnoreProperties(ignoreUnknown = true)
class CurrentWeather {
    public String description;
    public int temp;
    public int feels_like;
    public int wind_speed;
}

@RegisterRestClient(baseUri = "http://api.openweathermap.org/data/2.5")
interface WeatherService {

    enum MeasurementKind { standard, metric, imperial }

    // We can fake/pretend these are path params and it winds up a bit clearer
    @GET
    @Path("/weather?q={city},{country}&appid={apiKey}&units={units}")
    Set<CurrentWeather> getWeatherByCityAndCountry(String city,
                                                   String country,
                                                   String apiKey,
                                                   String units);
}

public class Thing {
    private static final Logger log = Logger.getLogger(SomePage.class);
    
    @RestClient WeatherService weatherApi;
    @Inject Configuration config;

    @GET
    @Path("/route")
    @Blocking
    public Set<CurrentWeather> getWeatherExample1() {
        log.info("called!!!");
        var results = weatherApi.getWeatherByCityAndCountry(
                "Miami", "US",
                config.openWeatherMapApiKey,
                WeatherService.MeasurementKind.imperial.name());
        log.info("results = ");
        log.info(results);
        return results;
    }
}
```

Nothing is ever logged past the initial `called!!!` 🙁 
Very confusing

image

image

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

Quarkus `v2.1.2-final`
```kts
plugins {
    java
    id("io.quarkus")
}

repositories {
    mavenCentral()
    mavenLocal()
}

val quarkusPlatformGroupId: String by project
val quarkusPlatformArtifactId: String by project
val quarkusPlatformVersion: String by project

dependencies {
    implementation(enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}"))
    implementation("io.quarkus:quarkus-arc")
    implementation("io.quarkus:quarkus-resteasy-reactive")
    implementation("io.quarkus:quarkus-resteasy-reactive-qute")
    implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
    implementation("io.quarkus:quarkus-rest-client-reactive")
    implementation("io.quarkus:quarkus-rest-client-reactive-jackson")

    testImplementation("io.quarkus:quarkus-junit5")
    testImplementation("io.rest-assured:rest-assured")
}

group = "org.acme"
version = "1.0.0-SNAPSHOT"

java {
    sourceCompatibility = JavaVersion.VERSION_16
    targetCompatibility = JavaVersion.VERSION_16
}

tasks.withType<JavaCompile> {
    options.encoding = "UTF-8"
    options.compilerArgs.add("-parameters")
}
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by stuartwdouglas**
If a call fails on the server it should be treated as an internal server
error, it should not be propagating the response back to the original
request.

Fixes #19488

### CODE REVIEW

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java [hunk 1]
```diff
@@ -115,6 +117,17 @@ public void abort() {
         restart(abortHandlerChain);
     }
 
+    @Override
+    protected Throwable unwrapException(Throwable t) {
+        var res = super.unwrapException(t);
+        if (res instanceof WebApplicationException) {
+            WebApplicationException webApplicationException = (WebApplicationException) res;
+            return new ClientWebApplicationException(webApplicationException.getMessage(), webApplicationException,
```
Comment (gastaldi) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java:125]: The `WebApplicationException` cause should be propagated here

Comment (michalszynkiewicz) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java:125]: I think it should.

I'm only wondering if it wouldn't be slightly better if `webApplication.getCause()` was passed as the cause and not `webApplication`

**Review by michalszynkiewicz — APPROVED** [review]

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java (hunk 1 above)
Comment (stuartwdouglas) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java:125]: I'm not sure, it does loose a bit of context as to where the original exception came from (e.g. a filter could directly throw it with no cause.)

**Review by geoand — APPROVED** [review]

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java (hunk 1 above)
Comment (michalszynkiewicz) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java:125]: true, let's leave it

**Pull request merged**